utils/validator.go

```diff
@@ -37,24 +37,65 @@ func NewValidator() Validator {
 	return Validator{}
 }
 
+func (validator Validator) ValidateHostnames(hostnames []string) error {
+	var firstHostname = hostnames[0]
+	isWorker, err := regexp.Match(`^ncn-w[0-9]*$`, []byte(firstHostname))
+	if err != nil {
+		return err
+	}
+	if isWorker {
```

Comment:
i don't think this if-else is needed here. what i'd do here:
```
        var firstHostname = hostnames[0]
        // worker
	isWorker, err := regexp.Match(`^ncn-w[0-9]*$`, []byte(firstHostname))
	if err != nil {
		return err
	}
	if isWorker {
	   return validator.ValidateWorkerHostnames(hostnames)
	}
        // storage
        isStorage, err := regexp.Match(`^ncn-s[0-9]*$`, []byte(firstHostname))
        if err != nil {
	        return err
        }
        if isStorage {
	        return validator.ValidateStorageHostnames(hostnames)
        }
        // master/xyz/whateverInTheFuture
        
        // nothing valid so we return error 
        return fmt.Errorf("invalid  hostname: %v", firstHostnames)
```

Reply:
good idea, it's changed

Reply:
hmm i don't see the change here, forget to git push?